code/headers/steering_wheel_controller.hpp

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include <controller_interface.hpp>
+#include <hwlib.hpp>
+#include <joystick_value.hpp>
+
+namespace r2d2::manual_control {
+
+    /**
+     * Class steering_wheel_controller_c provides functions
+     * for reading the data of the trust steering wheel.
+     */
+    class steering_wheel_controller_c : public hwlib::pin_in,
```

Comment:
Why are you inheriting from `hwlib::pin_in` ? you've made the `read` function private and it calls `refresh` on itself, so I'm not sure what is going on here...